install.sh

```diff
@@ -521,6 +536,90 @@ do_install() {
 			echo_docker_as_nonroot
 			exit 0
 			;;
+			
+		sles)
+			arch=$(uname -m)
+			if [[ $arch != "s390x" ]]; then
+				echo "Docker-CE is not supported on SLES for $arch!!!"
+				exit 1
+			else
+				echo "Installing Docker-CE on $arch!"
+			fi
+
+			sles_repo="$DOWNLOAD_URL/linux/$lsb_dist/$REPO_FILE"
+			opensuse_repo="https://download.opensuse.org/repositories/security:SELinux/SLE_15_SP2/security:SELinux.repo"
+			if ! curl -Ifs "$sles_repo" > /dev/null; then
+				echo "Error: Unable to curl repository file $sles_repo, is it valid?"
+				exit 1
+			fi
+			pre_reqs="ca-certificates curl libseccomp2 iptables awk"
+			
+			(
+				if ! is_dry_run; then
+					set -x
+				fi
+				$sh_c "zypper install -y $pre_reqs"
+				$sh_c "zypper addrepo $sles_repo"
+				
+				
+				cat >&2 <<-'EOF'
+						Warning !!! 
+						"Opensuse Repository (https://download.opensuse.org/repositories/security:SELinux) will be enabled now.
+						Do you wish to continue?
+				
+						You may press Ctrl+C now to abort this script.
+						EOF
+				( set -x; sleep 30 )
+				$sh_c "zypper addrepo $opensuse_repo"
+				$sh_c "zypper --gpg-auto-import-keys refresh"
+				$sh_c "zypper lr -d"
+				
+			)
+			pkg_version=""
+			if [ -n "$VERSION" ]; then
+				if is_dry_run; then
+					echo "# WARNING: VERSION pinning is not supported in DRY_RUN"
+				else
+					pkg_pattern="$(echo "$VERSION" | sed "s/-ce-/\\\\.ce.*/g" | sed "s/-/.*/g")"
+					search_command="zypper search -s --match-exact 'docker-ce' | grep '$pkg_pattern' | tail -1 | awk '{print \$6}'"
+					pkg_version="$($sh_c "$search_command")"
+					echo "INFO: Searching repository for VERSION '$VERSION'"
+					echo "INFO: $search_command"
+					if [ -z "$pkg_version" ]; then
+						echo
+						echo "ERROR: '$VERSION' not found amongst zypper list results"
+						echo
+						exit 1
+					fi
+					search_command="zypper search -s --match-exact 'docker-ce-cli' | grep '$pkg_pattern' | tail -1 | awk '{print \$6}'"
+					# It's okay for cli_pkg_version to be blank, since older versions don't support a cli package
+					cli_pkg_version="$($sh_c "$search_command")"
+					# Cut out the epoch and prefix with a '-'
+					pkg_version="-$(echo "$pkg_version")"
+					
+					search_command="zypper search -s --match-exact 'docker-ce-rootless-extras' | grep '$pkg_pattern' | tail -1 | awk '{print \$6}'"
+					rootless_pkg_version="$($sh_c "$search_command")"
+					rootless_pkg_version="-$(echo "$rootless_pkg_version")"
```

Comment:
echo is not needed here as well if we don't manipulate the version

```suggestion
					rootless_pkg_version="-$rootless_pkg_version"
```

Reply:
Updated the script with suggested change